### PR TITLE
feat(ci): publish provider-components.yaml on tag

### PR DESCRIPTION
**Pull request by rsmitty**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -24,7 +24,7 @@ RUN go vet ./pkg/... ./cmd/...
 RUN go test ./pkg/... ./cmd/... -coverprofile cover.out
 
 # Build the manager binary
-FROM generate AS build
```
Comment (rsmitty) [inline Dockerfile]: not sure it matters, but wanted to make sure the build happened from an image that got `go vet`ted and `go fmt`ted

File: Makefile [hunk 2]
```diff
@@ -29,23 +19,14 @@ deploy: manifests
 manifests:
 	go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go all
 	sed -i'' -e 's@^- manager_auth_proxy_patch.yaml.*@#&@' config/default/kustomization.yaml
+	sed -i'' -e 's@image: .*@image: '"${IMG}"'@' ./config/default/manager_image_patch.yaml
 	kustomize build config/default/ > provider-components.yaml
 	echo "---" >> provider-components.yaml
 	kustomize build vendor/sigs.k8s.io/cluster-api/config/default/ >> provider-components.yaml
 
-# Generate code
-generate:
```
Comment (rsmitty) [inline Makefile]: generate will be called by either docker-build or test steps inside the dockerfile, so this should no longer be needed.

File: Makefile [hunk 3]
```diff
@@ -1,22 +1,12 @@
...
 
-all: test manager
+all: test docker-build
 
 # Run tests
 test:
 	docker build . --target $@ -t $(REPO):test
 
-# Build manager binary
-manager: generate fmt vet
```
Comment (rsmitty) [inline Makefile]: Unneeded. This is essentially our build step in the dockerfile.

Comment (andrewrynhard) [reply Makefile]: We need this to build locally no?

File: Makefile (hunk 2 above)
Comment (andrewrynhard) [reply Makefile]: I think we need this to get the local files so that we can build locally with `go build`.

**Comment by rsmitty** [comment]
Updated to keep ability to build and generate locally.

**Pull request merged**